client/src/client_read.c

```diff
@@ -597,9 +597,67 @@ int process_gfid_reads(unifyfs_client* client,
     /* this records the pointer to the temp request array if
      * we allocate one, we should free this later if not NULL */
     read_req_t* reqs = NULL;
+    if (client->use_node_local_extents) {
+        extents_list_t list = calloc(1, sizeof(struct extents_list));
+        struct extents_list* cur = list;
+        int num_request_selected = 0;
+        for (int i = 0; i < in_count; ++i) {
+            int fid = unifyfs_fid_from_gfid(client, in_reqs[i].gfid);
+            /* get meta for this file id */
+            unifyfs_filemeta_t* meta = unifyfs_get_meta_from_fid(client, fid);
+            if (!meta->attrs.is_laminated || !meta->needs_extents_sync) {
+                /* do not proceed for this request as
+                 * it is not a laminated file or has already been synced.*/
+                continue;
+            }
+            num_request_selected++;
+            off_t filesize_offt = unifyfs_gfid_filesize(client,
+                                                        in_reqs[i].gfid);
+            cur->value.file_pos = 0;
+            cur->value.length = filesize_offt - 1;
+            cur->value.gfid = in_reqs[i].gfid;
+            if (i < in_count - 1) {
+                cur->next = calloc(1, sizeof(struct extents_list));
+                cur->next->next = NULL;
+                cur = cur->next;
+            } else {
+                cur->next = NULL;
+            }
+            meta->needs_extents_sync = 0;
+        }
+        if (num_request_selected > 0) {
+            /* There are files which are laminated and
+             * require sync of extents */
+            int extent_count = 0;
+            extents_list_t extents = NULL;
+            int ret =
```

Comment:
redefinition of 'int ret'

Reply:
perhaps use 'int rc' instead. conventionally, 'ret' should be the function return value.

Reply:
I fixed this check my comment below